src/main/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStepExecution.java

```diff
@@ -58,6 +76,7 @@
     public Void run() throws Exception {
         if (step.getLabel() != null) {
             node.addAction(new LabelAction(step.getLabel()));
+            node.addAction(new MilestoneAction(step.getLabel()));
```

Comment:
Could be collapsed into one action `extend`ing `LabelAction` and `implements`ing `StageAction`.